src/udisks2monitor.cpp

```diff
@@ -651,6 +671,44 @@ void UDisks2::Monitor::doFormat(const QString &devicePath, const QString &dbusOb
     });
 }
 
+void UDisks2::Monitor::doSetLabel(const QString &devicePath, const QString &dbusObjectPath, const QVariantList &arguments)
+{
+    const QString dbusMethod = UDISKS2_BLOCK_SETLABEL;
+    if (devicePath.isEmpty()) {
+        qCCritical(lcMemoryCardLog) << "Cannot" << dbusMethod.toLower() << "without device name";
+        return;
+    }
+
+    QDBusMessage method = QDBusMessage::createMethodCall(
+                UDISKS2_SERVICE,
+                dbusObjectPath,
+                UDISKS2_FILESYSTEM_INTERFACE,
+                dbusMethod);
+    method.setArguments(arguments);
+    QDBusPendingCall pendingCall = QDBusConnection::systemBus().asyncCall(method);
+    QDBusPendingCallWatcher *watcher = new QDBusPendingCallWatcher(pendingCall, this);
+    connect(watcher, &QDBusPendingCallWatcher::finished,
+            this, [this, devicePath, dbusMethod](QDBusPendingCallWatcher *watcher) {
+        if (watcher->isError()) {
+            QDBusError error = watcher->error();
+            QByteArray errorData = error.name().toLocal8Bit();
+            const char *errorCStr = errorData.constData();
+            qCWarning(lcMemoryCardLog) << dbusMethod << "error:" << errorCStr;
+
+            for (uint i = 0; i < sizeof(dbus_error_entries) / sizeof(ErrorEntry); i++) {
+                if (strcmp(dbus_error_entries[i].dbusErrorName, errorCStr) == 0) {
+                    emit setLabelError(dbus_error_entries[i].errorCode);
+                    break;
+                }
+            }
+        }
+
+        watcher->deleteLater();
+    });
+
+    emit status(devicePath, Partition::Renaming);
```

Comment:
This is now the only new status change on the last update. Per se good, but what's now removing the Renaming state when the operation is done?

Reply:
Ok, I see. What status should be set when operation is done? "Renamed"?

Reply:
I'd expect Unmounted if the operation can be done only to unmounted partitions.

Reply:
Done.